### ISSUE TITLE
Add momentum factor into UpdateWarps operation

### ISSUE THREAD

**Issue by Algomorph** [Algomorph/InfiniTAM#186]
weight_gradient + weight_momentum = 1.0
warp_update = weight_gradient * learning_rate * gradient + weight_momentum * warp_update
